iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp

```diff
@@ -44,6 +44,10 @@ namespace {
 
 struct ConvertHLOToLinalgOnTensorsPass
     : public PassWrapper<ConvertHLOToLinalgOnTensorsPass, FunctionPass> {
+  void getDependentDialects(DialectRegistry &registry) const override {
```

Comment:
The clang-format complaint is on this line - it wants

```
-  void getDependentDialects(DialectRegistry &registry) const override {
+  void getDependentDialects(DialectRegistry& registry) const override {
```

we're using the `DerivePointerAlignment` option (see https://clang.llvm.org/docs/ClangFormatStyleOptions.html) [1], which chooses the most common alignment of `&` and `*` in the file and sticks with that. In this case, we _should_ be using the style you have here (as this is compiler code, roughly following LLVM style), but the rest of the file uses the other alignment: 
https://github.com/google/iree/blob/6bd3b1dc13236566d1f8ee651cd3ab9b03dd8a1d/iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp#L58-L59
https://github.com/google/iree/blob/6bd3b1dc13236566d1f8ee651cd3ab9b03dd8a1d/iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp#L75-L76

You can either listen to clang-format here, or fix the rest of the file (preferably as part of a separate commit... but whatever, it's a small change :D)

[1] We could actually clean this up (@GMNGeoffrey) - We're technically using `BasedOnStyle: Google`: https://github.com/google/iree/blob/6bd3b1dc13236566d1f8ee651cd3ab9b03dd8a1d/.clang-format#L15-L18 which [sets that option](https://github.com/llvm/llvm-project/blob/bda8fbe2d2af208181efcf029b0c0bc0606da39b/clang/lib/Format/Format.cpp#L981), while the LLVM style [does not set the option](https://github.com/llvm/llvm-project/blob/bda8fbe2d2af208181efcf029b0c0bc0606da39b/clang/lib/Format/Format.cpp#L881). The cleanup would be to disable the internal tooling which has those issues now that we're developing directly on GitHub, then add `BasedOnStyle: LLVM` under `iree/compiler`.

Reply:
I think we can't really switch to `BasedOnStyle: LLVM` because we still format with google formatting on import and export (we can work towards not needing to do that). This particular issue has come up a couple times though. Since Google style just derives from the rest of the file, we could enforce a specific alignment (probably following LLVM makes sense) to avoid mismatch across the project.

Reply:
> because we still format with google formatting on import and export (we can work towards not needing to do that)

Yeah, that's what I'm suggesting. That formatting is also causing extra diffs in markdown and python files (which aren't really worth the trouble) when merging google -> main.

Reply:
Yeah but that will take a bit. Not formatting C++ will also mess up includes. So if we care about the pointer alignment specifically we could fix that without all the fixes necessary to turn off migration formatting entirely.

Reply:
Okay, fixed with #3008, which was merged into main. And thanks a lot @ScottTodd for the detailed response, I realy appreciate this. I wasn't aware of the `DerivePointerAlignment` clang-format setting.